miniapps/navier/navier_turbchan.cpp

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2010-2021, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "navier_solver.hpp"
+#include <fstream>
+#include <cmath>
+
+using namespace mfem;
+using namespace navier;
+
+struct s_NavierContext
+{
+   int order = 5;
+   double Re_tau = 180.0;
+   double kin_vis = 1.0 / Re_tau;
+   double t_final = 50.0;
+   double dt = -1.0;
+} ctx;
+
+double mesh_stretching_func(const double y)
+{
+   double C = 1.8;
+   double delta = 1.0;
+
+   return delta * tanh(C * (2.0 * y - 1.0)) / tanh(C);
+}
+
+void accel(const Vector &x, double t, Vector &f)
+{
+   f(0) = 1.0;
+   f(1) = 0.0;
+   f(2) = 0.0;
+}
+
+void vel_ic_reichardt(const Vector &coords, double t, Vector &u)
+{
+   double yp;
+   double x = coords(0);
+   double y = coords(1);
+   double z = coords(2);
+
+   double C = 5.17;
+   double k = 0.4;
+   double eps = 1e-2;
+
+   if (y < 0)
+   {
+      yp = (1.0 + y) * ctx.Re_tau;
+   }
+   else
+   {
+      yp = (1.0 - y) * ctx.Re_tau;
+   }
+
+   u(0) = 1.0 / k * log(1.0 + k * yp) + (C - (1.0 / k) * log(k)) * (1 - exp(
+                                                                       -yp / 11.0) - yp / 11.0 * exp(-yp / 3.0));
+
+   double kx = 23.0;
+   double kz = 13.0;
+
+   double alpha = kx * 2.0 * M_PI / 2.0 * M_PI;
+   double beta = kz * 2.0 * M_PI / M_PI;
+
+   u(0) += eps * beta * sin(alpha * x) * cos(beta * z);
+   u(1) = eps * sin(alpha * x) * sin(beta * z);
+   u(2) = -eps * alpha * cos(alpha * x) * sin(beta * z);
+}
+
+void vel_wall(const Vector &x, double t, Vector &u)
+{
+   u(0) = 0.0;
+   u(1) = 0.0;
+   u(2) = 0.0;
+}
+
+int main(int argc, char *argv[])
+{
+   MPI_Session mpi(argc, argv);
+
+   Device("ceed-cpu");
```

Comment:
Is this necessary? Related note: it is probably best to include a note saying this is a very resource intensive miniapp and requires around X hours on Y cpus to get meaningful results?

Reply:
It's not necessary. I can use a different backend by default.